### PR TITLE
#WIP System Admin Views for the following functionalities.

### PR DESCRIPTION
**Pull request by katafira**
<h1>Purpose or Intent</h1>

System Admin Views for the following functionalities.

Adding the capability for the administrator to stop changes in users. No new users will be added, no offspring, shifts won't be able to be changed or reassigned, except for the administrator. Users can delete their accounts in any case.
Switch system on/off: being able to close the system (so no new users can be added), until the admin changes a switch, so we can deploy before the date an hour when the system is open
Lock in of database, so there is no way to assing more offsprings than the available space.
Fix patternfly.
Fix views and duplicated code. 

@alexvkcr @delaluzparra @sergio-ocon

### CODE REVIEW

**Comment by delaluzparra** [comment]
Title and body in EN, please.

**Comment by sergio-ocon** [comment]
It would be great if for this kind of use cases you upload an image or gif (Álvaro can help with that)

**Comment by katafira** [comment]
@sergio-ocon @delaluzparra 

What was done to make bower jquery work:

1. Updated patternfly (bower)
2. Edited the following line in /config/application.rb
                     config.assets.paths << Rails.root.join('vendor', 'assets', 'bower_components')
3. Added the following lines in app/assets/javascripts/application.js
                     //= require_self
                     //= require bootstrap-switch
                     //= require datatables
What was done to check if query actually works:

1. Removed non working scripts.
2. Added buttons in system administration view
3. Checked table views and done some changes so that everything would fit properly with patternfly. Offspring table now has a similar structure as User table. (/app/views/offsprings/_offspring.html.erb)        


All these changes were made in the same pull request due to a secondary effect of fixing bower (patternfly). Tables now show search filters and have a better visual structure. 
<h3>Offspring table view structure was changed to be similar to the User tables so that it could be compatible with patternfly.</h3>
Make sure that the bower components paths are correct in app/views/layouts/application.html.erb:
            <%= render 'layouts/shim' %>
            
            <link href="bower_components/patternfly/dist/css/patternfly.min.css" rel="stylesheet" media="screen, print">
            <link href="bower_components/patternfly/dist/css/patternfly-additions.min.css" rel="stylesheet" media="screen, print">
            
            <script src="bower_components/jquery/dist/jquery.min.js"></script>
            
            <script src="bower_components/bootstrap/dist/js/bootstrap.min.js"></script>
            
            <script src="bower_components/patternfly/dist/js/patternfly.min.js"></script> 

Authors: @katafira @alexvkcr

File: vendor/assets/bower_components/angular-cookies/.bower.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (sergio-ocon) [inline vendor/assets/bower_components/angular-cookies/.bower.json]: This seems to be part of Angular, and we are not yet using AngularJS.... It creates a lot of noise in this PR.

One: if it is not needed, configure it out.
Two: if it is needed, create another PR

**Comment by sergio-ocon** [comment]
this PR does several things:

Fixes the javascript errors
updates bower components
Then do some things.

Please split it into meaningful PR, each one doing one single thing

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -11,12 +11,13 @@
 // about supported directives.
 //
 //= require jquery
-
-
-//
+//= require jquery_ujs
+//= require turbolinks
+//= require_tree .
```
Comment (sergio-ocon) [inline app/assets/javascripts/application.js]: In the bootstrap documentation is explicitly said that you don't use this...

Is it needed?

File: vendor/assets/bower_components/angular-cookies/.bower.json (hunk 1 above)
Comment (sergio-ocon) [inline vendor/assets/bower_components/angular-cookies/.bower.json]: My mistake. This is part of the angular-cookies component.

That is the reason why you split PR into pieces, in 1104 files is not possible to understand what is added by you and what by bower

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (katafira) [inline app/assets/javascripts/application.js]: It is needed. Without it scripts form bower are not accessible.

**Comment by katafira** [comment]
Closed because it has been split into various prs.

**Pull request closed without merging**